app/forms/questionnaire_form.py

```diff
@@ -328,28 +341,32 @@ def _option_value_in_data(answer, option, data):
     return option["value"] in dict(data_to_inspect).get(answer["id"], [])
 
 
-def get_answer_fields(question, data, error_messages, answer_store, metadata, location):
+def get_answer_fields(
+    question, data, schema, answer_store, list_store, metadata, location
+):
     answer_fields = {}
     question_title = question.get("title")
     for answer in question.get("answers", []):
         for option in answer.get("options", []):
             if "detail_answer" in option:
                 disable_validation = not _option_value_in_data(answer, option, data)
                 detail_answer = option["detail_answer"]
+
                 answer_fields[option["detail_answer"]["id"]] = get_field_handler(
                     detail_answer,
-                    error_messages,
+                    schema,
```

Comment:
I think we should consider initialising the `ValueSourceResolver` before looping over the answers, and passing it to the `get_field_handler` method. It looks like that would avoid the need to pass schema and the various stores (answers, lists, metadata). 

It looks like the only use of a store outside of resolving value sources is for calculations (https://github.com/ONSdigital/eq-questionnaire-runner/blob/refactor-field-handler/app/forms/questionnaire_form.py#L110), which could be modified to use the value source resolver. For now this could be done in code rather than modifying the `calculations` schema and another card raised to make the schema changes.

Reply:
:hear_no_evil: I would have to change those bloody tests again

Reply:
Yeah, not ideal, but a lot of them could be reverted, and would end up simpler.